ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -262,19 +353,10 @@ func validateRayCluster(admissionReview *admissionv1.AdmissionReview) (*admissio
 	workerGroupSpecs := raycluster.Spec.WorkerGroupSpecs
 	for i := 0; i < len(workerGroupSpecs); i++ {
 		workerGroupSpec := workerGroupSpecs[i]
-		if containerRequestingTPUs(workerGroupSpec.Template.Spec.Containers...) {
-			klog.V(1).InfoS("validateRayCluster", "RayCluster", namespace+"/"+clusterName, "Worker Group", workerGroupSpec.GroupName, "Requests TPUs", true)
-			replicas := int(*workerGroupSpec.Replicas)
-			numOfHosts := workerGroupSpec.NumOfHosts
-			for replicaIndex := 0; replicaIndex < replicas; replicaIndex++ {
-				// reset past sliceToWorkers entries for slice in ray cluster
-				groupName := workerGroupSpec.GroupName
-				podSlice := slice{clusterName, groupName, namespace, replicaIndex, numOfHosts}
-				sliceToWorkers[podSlice] = nil
-			}
-		} else {
-			// RayCluster worker group does not request TPUs
-			klog.V(1).InfoS("validateRayCluster", "RayCluster", namespace+"/"+clusterName, "Worker Group", workerGroupSpec.GroupName, "Requests TPUs", false)
+		workerGroupContainers := workerGroupSpec.Template.Spec.Containers
+		if len(workerGroupContainers) != 0 && !containerRequestingTPUs(workerGroupContainers...) {
+			// pass through if no TPUs are requested
+			break
```

Comment:
Should this be a `break` or `continue`? This means if the first worker group doesn't request TPUs, but the second one does, we would miss validation for the second

Reply:
Yeah it should be a continue, I missed that. Fixed in [fdf4e6c](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/fdf4e6c2c41d73bc980f99256b8c39f308205860).